pkg/transformer/kubernetes/kubernetes.go

```diff
@@ -463,6 +463,10 @@ func (k *Kubernetes) Transform(komposeObject kobject.KomposeObject, opt kobject.
 		service := komposeObject.ServiceConfigs[name]
 		var objects []runtime.Object
 
+		// Show error if no controller object is specified and restart: "on-failure"
+		if opt.CreateD && service.Restart == "on-failure" || opt.CreateDS && service.Restart == "on-failure" || opt.CreateDeploymentConfig && service.Restart == "on-failure" || opt.CreateRC && service.Restart == "on-failure" {
```

Comment:
this doesn't look right :-(
It should be also handled somewhere else. Don't forget about OpenShift transformer ;-)